### PR TITLE
Updated MR to Add serializable support

### PR DESCRIPTION
**Pull request by samskiter**

### CODE REVIEW

**Comment by samskiter** [comment]
Hoorah! Fixed and working hopefully!

**Comment by joelittlejohn** [comment]
@samskiter Great, thanks!

Could you move some of the 'serializable' stuff out of the ObjectRule into a separate helper? There's a lot of code in here already and there is a big block of code that is _only_ related to adding serialization support. I'd rather move all this (and use of the `java.security` classes) out into a separate helper. This is similar to what has already been done for Parcelable support.

Another minor thing, I notice your code style doesn't fit the rest of the code base regarding alignment declarations and assignments. Could you avoid creating the columns seen here:

https://github.com/joelittlejohn/jsonschema2pojo/pull/543/files#diff-7fef584364765f2bd47a596ad18cecc4R293

Cheers!

**Comment by samskiter** [comment]
@joelittlejohn Thanks for adding me to the repo. I'm still not sure I know what I'm doing with the build system, so I'll probably avoid pushing anything just yet! This PR is still broken, for example!

**Comment by samskiter** [comment]
Whoops, cross talk! I'll try to address your points as I fix things up.

**Comment by joelittlejohn** [comment]
@samskiter Yeah, no problem, hope you don't mind me adding you. Probably still best to submit changes as PRs so that we can discuss them. If there are _very_ minor things (e.g. fixing comments or other non-functional tweaks) you'd just like to push to master then feel free.

**Comment by samskiter** [comment]
@joelittlejohn OK, I think we could be there. Needs a careful look over obviously - I'm sure there could be some areas I've screwed up and debugging any issues resulting from this this would be really hard!

File: jsonschema2pojo-core/src/main/java/org/jsonschema2pojo/util/SerializableHelper.java [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+
+            for (int i = Math.min(digestBytes.length, 8) - 1; i >= 0; i--) {
+                serialVersionUID = serialVersionUID << 8 | (long)(digestBytes[i] & 0xff);
+            }
+
+            JFieldVar  serialUIDField = jclass.field(JMod.PRIVATE | JMod.STATIC | JMod.FINAL, long.class, "serialVersionUID");
+            serialUIDField.init(JExpr.lit(serialVersionUID));
+
+        } catch (IOException exception) {
+            final InternalError internalError = new InternalError(exception.getMessage());
```
Comment (joelittlejohn) [inline jsonschema2pojo-core/src/main/java/org/jsonschema2pojo/util/SerializableHelper.java]: You can use:

```java
throw new GenerationException(exception);
```

here.

File: jsonschema2pojo-core/src/main/java/org/jsonschema2pojo/util/SerializableHelper.java [hunk 2]
```diff
@@ -0,0 +1,180 @@
...
+            internalError.initCause(exception);
+            throw internalError;
+        } catch (NoSuchAlgorithmException exception) {
+            final RuntimeException securityException = new SecurityException(exception.getMessage());
```
Comment (joelittlejohn) [inline jsonschema2pojo-core/src/main/java/org/jsonschema2pojo/util/SerializableHelper.java]: And here.

File: jsonschema2pojo-core/src/main/java/org/jsonschema2pojo/rules/ObjectRule.java [hunk 3]
```diff
@@ -164,6 +169,10 @@ private void addParcelSupport(JDefinedClass jclass) {
         parcelableHelper.addCreator(jclass);
     }
 
+    private static void addSerializableSupport(JDefinedClass jclass) {
```
Comment (joelittlejohn) [inline jsonschema2pojo-core/src/main/java/org/jsonschema2pojo/rules/ObjectRule.java]: This method seems redundant. I'd inline it.

**Comment by joelittlejohn** [comment]
Closes #404.

**Comment by joelittlejohn** [comment]
@samskiter If you could make the tweaks above I'll merge this for the next release.

**Comment by joelittlejohn** [comment]
Also, feel free to squash this.

**Comment by samskiter** [comment]
Squashed and rebased to here: https://github.com/joelittlejohn/jsonschema2pojo/pull/549

**Pull request closed without merging**